slang.h

```diff
@@ -891,6 +891,7 @@ extern "C"
             @return The function pointer related to the name or nullptr if not found 
             */
         virtual SLANG_NO_THROW SlangFuncPtr SLANG_MCALL findFuncByName(char const* name) = 0;
+        virtual SLANG_NO_THROW void* SLANG_MCALL findObjectByName(char const* name) = 0;
```

Comment:
There's a couple issues I want to note here.

The first issue is that `ISlangSharedLibrary` is not just an interface that Slang vends out to applications, but it is also an interface that application code can use when passing things into Slang. I believe we have APIs that let the user say "here is the implementation of dxcompiler I want you to use" and the user passes in an `ISlangSharedLibrary`. Simply adding new requirements to interfaces that get passed in the app->Slang direction is a binary-incompatible change, so we need to be careful about it.

The second issue is that the naming for `findObjectByName` makes it a little tricky for the user to figure out what it does. Is it related to "objects" in the C++ sense? Object files? The lack of documentation here doesn't help, but I assume the intention was to piggyback on the documentation for `findFuncByName`, since they really represent the same operation.

That last point brings me to what I *think* is a reasonable solution:

We can actually just replace the `findFuncByName` operation entirely, and instead have a single `findSymbolByName` (or maybe `findSymbolAddressByName`) operation, which is intended to work for both functions and data.

Because the new operation would be replacing the old one, the change will be binary-compatible on any platform where function pointers and data pointers are the same size (which is a constraint I'm willing to live with since all platforms we are likely to care about have that guarantee).

We can then provide `findFuncByName` as an inline wrapper function so that existing client code continues to work.

Th change would be binary-compatible but source-breaking for any client code that defined their own `ISlangSharedLibrary`, but I can live with that.

A more extreme alternative is to leave `findFuncByName` as-is, and simply introduce a `findSymbolByName` that wraps it and returns a `void*` for the cases where somebody wants to look up a non-function symbol.

Reply:
Renamed `findFuncByName` to `findSymbolAddressByName` and provided an inline wrapper for `findFuncByName`.